include/swift/AST/DiagnosticsSema.def

```diff
@@ -1902,6 +1902,12 @@ ERROR(type_does_not_conform_owner,none,
 ERROR(type_does_not_conform_in_decl_ref,none,
       "referencing %0 %1 on %2 requires that %3 conform to %4",
       (DescriptiveDeclKind, DeclName, Type, Type, Type))
+ERROR(contextual_member_ref_on_protocol_requires_self_requirement,none,
+      "contextual member reference to %0 %1 requires "
```

Comment:
It might be better to use the term 'implicit member reference' because I think that's what they're officially called in TSPL (no need to change this now - we can always bikeshed diagnostics in a follow-up PR)